### DISCUSSION TITLE
[HELP] Setting up Coilpack Auth with Nuxt

### DISCUSSION THREAD

**Discussion by Zignature** [ExpressionEngine/Coilpack#17] [Q&A]
Hi folks,

I have a Nuxt app that uses GraphQL to consume the data from my EE backend. The app is not supposed to be publicly accessible, so I need authentication.

Eventually I'd like to build a CRUD app for the EE backend.

Being a novice when it comes to Nuxt and certainly when it comes to Laravel and Coilpack, I could use help setting up Auth in my Nuxt app. I can't see the forest for the trees.

I only want to use ExpressionEngine authentication, so it is clear to me I have to set `coilpack` as the default authentication guard. 

I will be using [nuxt-auth-sanctum](https://github.com/manchenkoff/nuxt-auth-sanctum).

What is the endpoint to query for the user? 

Any help would be greatly appreciated.

**Comment by bryannielsen** [comment]
@Zignature I haven't tested this but I think what you need to do is setup a new Auth guard that uses `sanctum` as the driver and `coilpack` as the provider.

So in your Laravel app's `config/auth.php` file you would add something like

```
...
'guards' => [
   ...
    'my-new-guard' => [
        'driver' => 'sanctum',
        'provider' => 'coilpack',
    ],
    ...
],
...
```

And then to create your user endpoint it would look something like this in your `routes.php` file

```
Route::middleware(['auth:my-new-guard'])->get('/user', function(){
    return auth()->guard('my-new-guard')->user();
}
```

If I get some time I'll try to setup Sanctum and test this out but maybe someone with more experience will chime in before then :-)

**Reply by Zignature** [reply]
@bryannielsen thank you so much!

By the way, I'm not bound by `nuxt-auth-sanctum`. Any other way to authenticate with EE credentials is fine by me.
I just figured Sanctum was the way to go because Sanctum is in `composer.json`. I could use Laravel Breeze as well, or any other interface you could recommend.

**Comment by bryannielsen** [comment]
@Zignature I went through getting setup with Sanctum and wrote a guide here - https://expressionengine.github.io/coilpack-docs/docs/guides/using-laravel-sanctum would you mind reviewing that for me and let me know if anything doesn't make sense or line up with your use case?

I think Sanctum is a great choice for what you're looking to do, hopefully this helps get you going!

**Reply by Zignature** [reply]
@bryannielsen that's awesome! 

I'll have a look at it later today. It's 8AM here right now :)
